### PR TITLE
add checkout depth1 to workflow to access local github actions like generate allure report

### PR DESCRIPTION
**Pull request by Bodobolero**
## Problem

job step to create allure report fails

https://github.com/neondatabase/neon/actions/runs/9781886710/job/27006997416#step:11:1

## Summary of changes

Shallow checkout of sources to get access to local github action needed in the job step

## Example run 
example run with this change https://github.com/neondatabase/neon/actions/runs/9790647724
do not merge this PR until the job is clean

### CODE REVIEW

**Review by bayandin — COMMENTED** [review]
Just to clarify, to create an allure report by  `./.github/actions/allure-report-generate` we need to upload allure results (from tests) by `./.github/actions/allure-report-store` first.
For the regular tests/benchmarks it is done in `.github/actions/run-python-test-set`, do we do the same for Periodic pagebench?

**Comment by Bodobolero** [comment]
>  do we do the same for Periodic pagebench?

Yes we do here https://github.com/neondatabase/ec2_test_runner/blob/main/pagebench/upload_allure_report.sh

File: .github/workflows/periodic_pagebench.yml [hunk 1]
```diff
@@ -43,6 +43,14 @@ jobs:
       AWS_DEFAULT_REGION : "eu-central-1"
       AWS_INSTANCE_ID : "i-02a59a3bf86bc7e74"
     steps:
+    # we don't need the neon source code because we run everything remotely
+    # however we still need the local github actions to run the allure step below
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        submodules: false
+        fetch-depth: 1
```
Comment (bayandin) [inline .github/workflows/periodic_pagebench.yml:52]: Those are default args, we don't need to set them explicitly

```suggestion
```

File: .github/workflows/periodic_pagebench.yml [hunk 2]
```diff
@@ -43,6 +43,14 @@ jobs:
       AWS_DEFAULT_REGION : "eu-central-1"
       AWS_INSTANCE_ID : "i-02a59a3bf86bc7e74"
     steps:
+    # we don't need the neon source code because we run everything remotely
+    # however we still need the local github actions to run the allure step below
+    - name: Checkout
+      uses: actions/checkout@v4
```
Comment (bayandin) [inline .github/workflows/periodic_pagebench.yml:49]: I think the name is clear for this action here, so we can use just

```suggestion
    - uses: actions/checkout@v4
```

**Review by bayandin — APPROVED** [review]

**Review by bayandin — APPROVED** [review]

**Comment by Bodobolero** [comment]
successful run with allure report https://github.com/neondatabase/neon/actions/runs/9799530541

**Pull request merged**